Flaky test: `planbuilder` TestOperator

### Overview of the Issue

Looks like the test is expecting the columns to be in a consistent order, but that's not always true:

```
--- FAIL: TestOperator (0.00s)
139
    --- FAIL: TestOperator/455:update_tbl_set_col1_=_'apa',_col2_=_1337_where_id_=_12_and_name_=_'gangal'_ (0.00s)
140
        operator_test.go:110: 
141
            	Error Trace:	operator_test.go:110
142
            	Error:      	Not equal: 
143
            	            	expected: "Update {\n\ttable: \tTableSet{0}:tbl where id = 12 and `name` = 'gangal'\nassignments:\n\tcol1 = 'apa'\n\tcol2 = 1337\n}"
144
            	            	actual  : "Update {\n\ttable: \tTableSet{0}:tbl where id = 12 and `name` = 'gangal'\nassignments:\n\tcol2 = 1337\n\tcol1 = 'apa'\n}"
145
            	            	
146
            	            	Diff:
147
            	            	--- Expected
148
            	            	+++ Actual
149
            	            	@@ -3,4 +3,4 @@
150
            	            	 assignments:
151
            	            	+	col2 = 1337
152
            	            	 	col1 = 'apa'
153
            	            	-	col2 = 1337
154
            	            	 }
155
            	Test:       	TestOperator/455:update_tbl_set_col1_=_'apa',_col2_=_1337_where_id_=_12_and_name_=_'gangal'_
156
FAIL
157
FAIL	vitess.io/vitess/go/vt/vtgate/planbuilder/abstract	0.016s
```

[ci link](https://github.com/vitessio/vitess/runs/6082737687?check_suite_focus=true#step:8:138)

cc @vitessio/query-serving 

### Reproduction Steps

run the test in question a few times

### Binary Version

```sh
current `main`
```


### Operating System and Environment details

```sh
GitHub CI, also locally
```

--- a/go/vt/vtgate/planbuilder/abstract/operator_test.go
+++ b/go/vt/vtgate/planbuilder/abstract/operator_test.go
@@ -167,8 +167,14 @@ func testString(op Operator) string {
 	case *Update:
 		tbl := "table: " + op.Table.testString()
 		var assignments []string
-		for name, expr := range op.Assignments {
-			assignments = append(assignments, fmt.Sprintf("\t%s = %s", name, sqlparser.String(expr)))
+		// sort to produce stable results, otherwise test is flaky
+		keys := make([]string, 0, len(op.Assignments))
+		for k := range op.Assignments {
+			keys = append(keys, k)
+		}
+		sort.Strings(keys)
+		for _, k := range keys {
+			assignments = append(assignments, fmt.Sprintf("\t%s = %s", k, sqlparser.String(op.Assignments[k])))
 		}
 		return fmt.Sprintf("Update {\n\t%s\nassignments:\n%s\n}", tbl, strings.Join(assignments, "\n"))
 	}

